examples/lustre/test.bats

```diff
@@ -0,0 +1,76 @@
+load "${CHTEST_DIR}/common.bash"
+
+setup () {
+    scope full
+    prerequisites_ok lustre
+
+    if [[ $SLURM_JOB_ID ]]; then
+        if [[ -z $ch_lustre ]]; then 
+            pedantic_fail 'no lustre to bind mount'
+        fi
+    else
+        if [[ -z $ch_lustre ]]; then
+            skip 'no lustre to bind mount'
+        fi
+    fi
+
+    # Check lustre directory exists
+    if [[ ! -e $ch_lustre ]]; then
+        echo "${ch_lustre} does not exist"
+        exit 1
+    fi
+    # Check lustre directory is a directory
+    if [[ ! -d $ch_lustre ]]; then
+        echo "${ch_lustre} is not a directory"
+        exit 1
+    fi
+
+    binds=${ch_lustre}:/mnt/0
+    work_dir=/mnt/0/charliecloud_testing_directory
+    mkdir -p "${ch_lustre}/charliecloud_testing_directory"
+}
+
+@test "${ch_tag}/create dir" {
+    ch-run -b "$binds" "$ch_img" -- mkdir "${work_dir}/tryme"
+}
+
+@test "${ch_tag}/create file" {
+    ch-run -b "$binds" "$ch_img" -- touch "${work_dir}/test_create.txt"
+}
+
+
+@test "${ch_tag}/delete file" {
+    ch-run -b "$binds" "$ch_img" -- rm -f "${work_dir}/test_create.txt"
+}
+
+@test "${ch_tag}/write" {
+    ch-run -b "$binds" "$ch_img" -- sh -c "sleep 5 && echo hello | dd of=${work_dir}/test_write.txt"
+}
+
+@test "${ch_tag}/read" {
+    out=$(ch-run -b "$binds" "$ch_img" -- dd if="${work_dir}/test_write.txt" iflag=nocache 2> /dev/null)
+    if [ "$out" != hello ]; then
+        echo "Content of ${work_dir}/test_write.txt invalid" && exit 1
+    fi
+}
+
+@test "${ch_tag}/striping" {
+    ch-run -b "$binds" "$ch_img" -- mkdir "${work_dir}/set_stripes"
+    ch-run -b "$binds" "$ch_img" -- lfs setstripe -c 1 "${work_dir}/set_stripes"
+    one_stripe=$(ch-run -b "$binds" "$ch_img" -- lfs getstripe "${work_dir}/set_stripes/" | grep stripe_count | gawk '{ print $2}')
```

Comment:
I think you can use the `run` and `diff` idiom here too.

Reply:
This is further complicated by different Lustre systems having different default settings. There is no way to match all the output using the run and diff idiom here on every system.

We could:
- Not use the run diff idiom here
- Capture all output for the "default" then compare other output to it to make sure it has changed

Reply:
Since system defaults can change this won't really work here. We could set a stripe count, then get the stripe count using `run` then we could change the stripe count and make sure the diff fails.

Reply:
But i might be missing something. I can't think of another way to really compare the unpredictable output. Any thoughts?

Reply:
How about something like this:

```
mkdir foo
old=$(lfs getstripe --stripe-count foo)
echo "old: $stripe_ct_old"
new_expected=$(($old * 2))
echo "new expected: $new_expected"
new_actual=$(lfs getstripe --stripe-count foo)
echo "new actual: $new_actual"
[[ $new_expected -eq $new_actual ]]
```

I think you're right that the `diff` trick is overkill here. The test will fail if any of the `lfs` invocations fail.

You might want to set a variable or function which contains all the `ch-run` options you need in this script, to reduce repetition.